### PR TITLE
Add persist docs for the Update 6 release

### PR DESCRIPTION
**Pull request by praneesha**
## Purpose
Add persist docs for the Update 6 release.
> Fixes #

## Checklist

- [ ] **Page addition**
  - [ ] Add `permalink` to pages.

- [ ] **Page removal**
  - [ ] Remove entry from corresponding left nav YAML file.
  - [ ] Add `redirect_from` on the alternative page.
  - [ ] If no alternative page, add redirection on the `redirections.js ` file.

- [ ] **Page rename**
  - [ ] Add front-matter `redirect_from`.
  - [ ] Add front-matter `redirect_to:` (if applicable).

- [ ] **Page restrcuture**
  - [ ] Add `permalink` to pages.
  - [ ] Add front-matter `redirect_from`.
  - [ ] Add front-matter `redirect_to:` (if applicable).

### CODE REVIEW

File: swan-lake/featured-scenarios/manage-data-persistence-with-bal-persist.md [hunk 1]
```diff
@@ -0,0 +1,502 @@
+---
+layout: ballerina-building-a-data-service-with-bal-persist-left-nav-pages-swanlake
+title: Manage data persistence with bal persist
+description: Describe how to model your application data and use the tool to generate client APIs to access the data store.
+keywords: ballerina, data service, data store, database, in-memory, google sheets, REST, API
+permalink: /learn/manage_data_persistence-with-bal-persist/
+active: manage_data_persistence-with-bal-persist
+intro: This guide helps you understand the basics of the bal persist, which allows you to manage data persistence easily. This same data service is also written using the Ballerina SQL connectors, which required you to write SQL queries to perform CRUD operations against the DB servers. With the Bal persist feature, you only need to write the `Data Model`. Based on the model, the client object and record types are generated to interact with the data store.
```
Comment (anupama-pathirage) [inline swan-lake/featured-scenarios/manage-data-persistence-with-bal-persist.md:8]:
```suggestion
intro: This guide helps you understand the basics of the bal persist, which allows you to manage data persistence easily. This same data service is also written using the Ballerina SQL connectors, which require you to write SQL queries to perform CRUD operations against the DB servers. With the Bal persist feature, you only need to write the `Data Model`. Based on the model, the client object and record types are generated to interact with the data store.
```

File: swan-lake/featured-scenarios/manage-data-persistence-with-bal-persist.md [hunk 2]
```diff
@@ -0,0 +1,502 @@
...
+
+1. [Ballerina 2201.6.0 (Swan Lake)](/learn/install-ballerina/set-up-ballerina/) or greater
+2. A text editor
+>**Tip:** Preferably, <a href="https://code.visualstudio.com/" target="_blank">Visual Studio Code</a> with the
+<a href="https://wso2.com/ballerina/vscode/docs/get-started/install-the-extension/" target="_blank">Ballerina extension</a> installed.
+3. A command terminal
+
+## Understand the implementation
+
+This tutorial describes how to interact with the data store and perform operations against it using bal persist. This will cover the basic use case of creating, reading, updating, and deleting records on a data store in an organization. It also elaborates on how you can create an HTTP RESTful API using Ballerina that can be used to perform basic CRUD operations on the data store.
```
Comment (anupama-pathirage) [inline swan-lake/featured-scenarios/manage-data-persistence-with-bal-persist.md:30]: What is the correct term to refer to the feature ? `bal persist` or `Ballerina persist` (Can see the latter in below `Info` block.) We need to be consistent.

File: swan-lake/featured-scenarios/manage-data-persistence-with-bal-persist.md [hunk 3]
```diff
@@ -0,0 +1,502 @@
...
+   string last_name;
+   string email;
+   string phone;
+   time:Date hire_date;
+   string? manager_id;
+   string job_title;
+|};
+```
+
+This model is used to set up the underline datastore (e.g.: set up the tables in the underlying database), and also as the base for the generated client APIs
```
Comment (anupama-pathirage) [inline swan-lake/featured-scenarios/manage-data-persistence-with-bal-persist.md:110]:
```suggestion
This model is used to set up the underline datastore (e.g., set up the tables in the underlying database) and also as the base for the generated client APIs
```

File: swan-lake/featured-scenarios/manage-data-persistence-with-bal-persist.md [hunk 4]
```diff
@@ -0,0 +1,502 @@
...
+
+## Generate the client object, types, and scripts
+
+Now you can generate the client objects, types, and SQL scripts for your model by running the command in your terminal,
+
+```
+$ bal persist generate
+```
+
+This changes the Ballerina package like below,
```
Comment (anupama-pathirage) [inline swan-lake/featured-scenarios/manage-data-persistence-with-bal-persist.md:121]:
```suggestion
This changes the Ballerina package like below.
```
```suggestion
This changes the Ballerina package as follows.
```

File: swan-lake/featured-scenarios/manage-data-persistence-with-bal-persist.md [hunk 5]
```diff
@@ -0,0 +1,502 @@
...
+<a href="https://wso2.com/ballerina/vscode/docs/get-started/install-the-extension/" target="_blank">Ballerina extension</a> installed.
+3. A command terminal
+
+## Understand the implementation
+
+This tutorial describes how to interact with the data store and perform operations against it using bal persist. This will cover the basic use case of creating, reading, updating, and deleting records on a data store in an organization. It also elaborates on how you can create an HTTP RESTful API using Ballerina that can be used to perform basic CRUD operations on the data store.
+
+![Data Service Architecture](/learn/images/data-service-architecture.png "Data Service Architecture")
+
+>**Info:** In this tutorial, we use an in-memory data store for simplicity. However, the methodology described here can also be applied to work with MySQL and Google Sheets as the Ballerina persist currently offers support for these three data stores: in-memory tables, MySQL databases, and Google Sheets.
```
Comment (daneshk) [inline swan-lake/featured-scenarios/manage-data-persistence-with-bal-persist.md:34]:
```suggestion
>**Info:** In this tutorial, we use an in-memory data store for simplicity. However, the methodology described here can also be applied to work with MySQL and Google Sheets as the `bal persist` currently offers support for these three data stores: in-memory tables, MySQL databases, and Google Sheets.
```

File: swan-lake/featured-scenarios/manage-data-persistence-with-bal-persist.md [hunk 6]
```diff
@@ -0,0 +1,502 @@
...
+module = "rainier.store"
+```
+
+These configurations are referred to when generating client objects for the data model. The recommendation is not to change these values. If you want to change it, remove these configurations from the `Ballerina.toml` file and reinitialize the bal persist.
+
+The next step is to define your data model in the schema file at `persist/model.bal`.
+
+## Model your data
+
+The data model can be defined in the schema file inside persist directory. You can use the empty file added by the `persist init` command for this.
```
Comment (daneshk) [inline swan-lake/featured-scenarios/manage-data-persistence-with-bal-persist.md:89]:
```suggestion
The data model can be defined in the schema file inside the `persist` directory. You can use the empty file added by the `persist init` command for this.
```

File: swan-lake/learn-the-platform/ballerina-persist/overview.md [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+
+> Note: The Ballerina VSCode plugin provides validation and code actions for the persist model. So you can easily create the persist model using the Ballerina VSCode plugin.
+
+## Persist CLI
+
+The `bal persist` CLI is used to generate the client API for the data model. Based on the data store, additional configurable file and setup scripts are generated. For example, if you are using MySQL as the data store, the `persist_db_config.bal` file, and the `script.sql` script are generated. The `persist_db_config.bal` file is used to configure the MySQL database connection. The `script.sql` file is used to create the table in the MySQL database. The client API is generated in the `generated` directory in the project root directory.
+
+The `bal persist` has two main built-in CLI commands: `persist init` and `persist generate`. The `persist init` command is used to initialize bal persist in the Ballerina project. The `persist generate` command is used to generate the client API for the data model. Additionally, there is an experimental `persist migrate` command to generate the sql scripts for changing the table structure in the database when the data model is changed.
```
Comment (daneshk) [inline swan-lake/learn-the-platform/ballerina-persist/overview.md:37]:
```suggestion
The `bal persist` has two main built-in CLI commands: `persist init` and `persist generate`. The `persist init` command is used to initialize bal persist in the Ballerina project. The `persist generate` command is used to generate the client API for the data model. Additionally, there is an experimental `persist migrate` command to generate the SQL scripts for changing the table structure in the database when the data model is changed.
```

File: swan-lake/featured-scenarios/manage-data-persistence-with-bal-persist.md [hunk 8]
```diff
@@ -5,8 +5,7 @@ description: Describe how to model your application data and use the tool to gen
 keywords: ballerina, data service, data store, database, in-memory, google sheets, REST, API
 permalink: /learn/manage_data_persistence-with-bal-persist/
 active: manage_data_persistence-with-bal-persist
-intro: This guide helps you understand the basics of the bal persist, which allows you to manage data persistence easily. This same data service is also written using the Ballerina SQL connectors, which required you to write SQL queries to perform CRUD operations against the DB servers. With the Bal persist feature, you only need to write the `Data Model`. Based on the model, the client object and record types are generated to interact with the data store.
-
+intro: This guide helps you understand the basics of the bal persist, which allows you to manage data persistence easily. This same data service is also written using the Ballerina SQL connectors, which require you to write SQL queries to perform CRUD operations against the DB servers. With the Bal persist feature, you only need to write the `Data Model`. Based on the model, the client object and record types are generated to interact with the data store.
```
Comment (daneshk) [inline swan-lake/featured-scenarios/manage-data-persistence-with-bal-persist.md:8]:
```suggestion
intro: This guide helps you understand the basics of the `bal persist` feature, which allows you to manage data persistence easily. This same data service is also written using the Ballerina SQL connectors, which require you to write SQL queries to perform CRUD operations against the DB servers. With the `bal persist` feature, you only need to write the `Data Model`. Based on the model, the client object and record types are generated to interact with the data store.
```

File: swan-lake/featured-scenarios/manage-data-persistence-with-bal-persist.md [hunk 9]
```diff
@@ -80,13 +79,14 @@ datastore = "inmemory"
 module = "rainier.store"
 ```
 
-These configurations are referred to when generating client objects for the data model. The recommendation is not to change these values. If you want to change it, remove these configurations from the Ballerina.toml file and reinitialize the bal persist.
+
+These configurations are referred to when generating client objects for the data model. The recommendation is not to change these values. If you want to change it, remove these configurations from the `Ballerina.toml` file and reinitialize the bal persist.
```
Comment (daneshk) [inline swan-lake/featured-scenarios/manage-data-persistence-with-bal-persist.md:83]:
```suggestion
These configurations are referred to when generating client objects for the data model. The recommendation is not to change these values. If you want to change it, remove these configurations from the `Ballerina.toml` file and reinitialize the `bal persist`.
```

**Review by sm1990 — APPROVED** [review]

**Pull request merged**